rc/lsp.kak

```diff
@@ -342,9 +342,39 @@ have_kakoune_feature_filtertext = ${kak_opt_lsp_have_kakoune_feature_filtertext}
 " | eval "${kak_opt_lsp_cmd} --request") > /dev/null 2>&1 < /dev/null & }
 }}
 
+declare-option -hidden str lsp_completion_selection ""
+
 define-command -hidden lsp-completion-dismissed -docstring "Called when the completion pager is closed" %{
+    set-option window lsp_completion_selection %val{hook_param}
+    try %{
+        eval -draft %{
+            select %opt{lsp_completion_selection}
+        }
+    } catch %{
+        # Completion was cancelled, nothing was inserted
+        # Clear these so the following lines do nothing.
+        # This is to avoid runing them in the above try block, and silencing their errors
+        remove-hooks window lsp-completion-dismissed
+        set-option window lsp_completions_selected_item -1
```

Comment:
I'm pretty sure the selected item is always -1 in this case, so this looks redundant

Reply:
It isnt - this case happens when the completion menu is opened, something is selected (which set selected_item and added on-dismissed hooks), and then deselected (by <c-p> for example), and the menu is closed with nothing inserted. In that case, the selected_item will still hold a value, but needs to be cleared, as no dismiss hooks should run.

Reply:
You're right, that's a bug in the implementation of `completionItem/resolve`.